### PR TITLE
deprecate access_token_passthrough

### PR DESCRIPTION
**Pull request by asreehari-splunk**
**Description:** access_token_passthrough config is being deprecated in favor of enabling include_metadata

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: receiver/sapmreceiver/trace_receiver.go [hunk 1]
```diff
@@ -208,6 +208,12 @@ func newReceiver(
 	if err != nil {
 		return nil, fmt.Errorf("failed to marshal default response body for %v receiver: %w", params.ID, err)
 	}
+	
+	if config.AccessTokenPassthrough {
+		params.Logger.Warn("access_token_passthrough is deprecated. Please enable include_metadata.")
+	}
+
+
```
Comment (dmitryax) [inline receiver/sapmreceiver/trace_receiver.go:216]:
```suggestion

```

File: .chloggen/deprecate_access_token_passthrough.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Deprecate the use of `splunk.AccessTokenPassthroughConfig` config field in favor of using include_metadata to pass token in request context.
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: []
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: Deprecating the use of `splunk.AccessTokenPassthroughConfig` config field means that it is recommended to use the include_metadata option to pass the token in the request context.
```
Comment (dmitryax) [inline .chloggen/deprecate_access_token_passthrough.yaml:18]: Please also add a line that the batch processor should be updated as well, even if we will do it automatically. The same for the warning message

File: .chloggen/deprecate_access_token_passthrough.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: deprecation
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: sapmreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Deprecate the use of `splunk.AccessTokenPassthroughConfig` config field in favor of using include_metadata to pass token in request context.
```
Comment (crobert-1) [inline .chloggen/deprecate_access_token_passthrough.yaml:10]:
```suggestion
note: Deprecate the use of `splunk.AccessTokenPassthroughConfig` config field in favor of using `include_metadata` to pass token in request context.
```
Tiny nit

File: .chloggen/deprecate_access_token_passthrough.yaml (hunk 2 above)
Comment (crobert-1) [inline .chloggen/deprecate_access_token_passthrough.yaml:18]:
```suggestion
subtext: Deprecating the use of `splunk.AccessTokenPassthroughConfig` config field means that it is recommended to use the `include_metadata` option to pass the token in the request context.
```
Tiny nit

File: receiver/sapmreceiver/trace_receiver.go [hunk 4]
```diff
@@ -208,6 +208,12 @@ func newReceiver(
 	if err != nil {
 		return nil, fmt.Errorf("failed to marshal default response body for %v receiver: %w", params.ID, err)
 	}
+	
+	if config.AccessTokenPassthrough {
+		params.Logger.Warn("access_token_passthrough is deprecated. Please enable include_metadata.")
```
Comment (crobert-1) [inline receiver/sapmreceiver/trace_receiver.go:213]:
```suggestion
		params.Logger.Warn("access_token_passthrough is deprecated. Please enable include_metadata instead.")
```

File: receiver/sapmreceiver/config.go [hunk 5]
```diff
@@ -13,5 +13,6 @@ import (
 type Config struct {
 	confighttp.ServerConfig `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
 
+	// Deprecated: `access_token_passthrough` is deprecated. Please enable include_metadata.
```
Comment (crobert-1) [inline receiver/sapmreceiver/config.go:16]:
```suggestion
	// Deprecated: `access_token_passthrough` is deprecated. Please enable include_metadata instead.
```

**Review by crobert-1 — COMMENTED** [review]
Looks good overall, just some tiny wording/formatting nits. Thanks @asreehari-splunk!

File: .chloggen/deprecate_access_token_passthrough.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: []
```
Comment (crobert-1) [inline .chloggen/deprecate_access_token_passthrough.yaml:13]:
```suggestion
issues: [35330]
```

File: .chloggen/deprecate_access_token_passthrough.yaml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: |
+  access_token_passthrough is deprecated. |
+  Please enable include_metadata in the receiver and add the following config to the batch processor |
+  ```yaml
+  batch:
+    metadata_keys: [X-Sf-Token]
```
Comment (dmitryax) [inline .chloggen/deprecate_access_token_passthrough.yaml:23]:
```suggestion
    metadata_keys: [X-Sf-Token]
    ```
```

File: .chloggen/deprecate_access_token_passthrough.yaml [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  ```
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [35330]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: |
+  access_token_passthrough is deprecated. |
+  Please enable include_metadata in the receiver and add the following config to the batch processor |
```
Comment (dmitryax) [inline .chloggen/deprecate_access_token_passthrough.yaml:20]:
```suggestion
subtext: -|
  `access_token_passthrough` is deprecated.
  Please enable include_metadata in the receiver and add the following config to the batch processor:
```

File: receiver/sapmreceiver/config.go [hunk 9]
```diff
@@ -13,5 +13,11 @@ import (
 type Config struct {
 	confighttp.ServerConfig `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
 
+	// Deprecated: `access_token_passthrough` is deprecated.
+	// Please enable include_metadata in the receiver and add the following config to the batch processor |
```
Comment (dmitryax) [inline receiver/sapmreceiver/config.go:17]:
```suggestion
	// Please enable include_metadata in the receiver and add the following config to the batch processor:
```

File: receiver/sapmreceiver/config.go [hunk 10]
```diff
@@ -13,5 +13,11 @@ import (
...
+	// ```yaml
+	// batch:
+	//      metadata_keys: [X-Sf-Token]
+	// ```
```
Comment (dmitryax) [inline receiver/sapmreceiver/config.go:21]: markdown won't work here
```suggestion
	// batch:
	//    metadata_keys: [X-Sf-Token]
```

**Comment by dmitryax** [comment]
LGTM, other than the formatting nits. Thank you!

**Review by dmitryax — APPROVED** [review]

**Pull request merged**